dinky-admin/src/main/resources/mapper/AlertRulesMapper.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+<mapper namespace="org.dinky.mapper.AlertRulesMapper">
+
+
+    <select id="selectWithTemplate" resultType="org.dinky.data.dto.AlertRuleDTO">
+        select dar.id,
+               dar.name,
+               dar.enabled,
+               dar.rule,
+               dar.template_id,
+               dar.rule_type,
+               dar.trigger_conditions,
+               dar.description,
+               dat.name as template_name,
+               dat.template_content
+        from dinky_alert_rules dar
+                 left join dinky.dinky_alert_template dat on dar.template_id = dat.id
+        where dar.enabled = 1
+    </select>
+
+
+    <select id="selectForProTable" resultType="org.dinky.data.model.AlertRule">
+        select
+        a.*
+        from
+        dinky_alert_rules a
+        <where>
+            1=1
```

Comment:
'1=1' isn't needed after <where> statement.

Reply:
> '1=1' isn't needed after statement.

The `1 = 1` is needed here, follow-up `and `operation maybe does not exist

Reply:
[trim, where, set](https://mybatis.org/mybatis-3/dynamic-sql.html#:~:text=MyBatis%20has%20a%20simple%20answer%20that%20will%20likely%20work%20in%2090%25%20of%20the%20cases.%20And%20in%20cases%20where%20it%20doesn%E2%80%99t%2C%20you%20can%20customize%20it%20so%20that%20it%20does.%20With%20one%20simple%20change%2C%20everything)

Reply:
> [trim, where, set](https://mybatis.org/mybatis-3/dynamic-sql.html#:~:text=MyBatis%20has%20a%20simple%20answer%20that%20will%20likely%20work%20in%2090%25%20of%20the%20cases.%20And%20in%20cases%20where%20it%20doesn%E2%80%99t%2C%20you%20can%20customize%20it%20so%20that%20it%20does.%20With%20one%20simple%20change%2C%20everything)

Well, it's been deleted